src/hotspot/os/posix/os_posix.cpp

```diff
@@ -799,19 +799,19 @@ int os::socket_close(int fd) {
   return ::close(fd);
 }
 
-int os::recv(int fd, char* buf, size_t nBytes, uint flags) {
+ssize_t os::recv(int fd, char* buf, size_t nBytes, uint flags) {
   RESTARTABLE_RETURN_INT(::recv(fd, buf, nBytes, flags));
```

Comment:
Note the name of the macro: this is supposed to return int. We need a new macro for RESTARTABLE_RETURN_SSIZE_T

Reply:
I renamed the macro, since the callers all return size_t.